his_checker.py

```diff
@@ -0,0 +1,30 @@
+#!/bin/python3
+
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) Bao Project and Contributors. All rights reserved
+
+"""
+Generating HIS metrics check
+"""
+
+import sys
+import argparse
+
+METRIC_FAIL = 0
+
+def process_metric(filenames):
+
+    """Process each file."""
+    print("Checking X metric")
+    for file in filenames.files:
+        print(file)
+
+if __name__ == "__main__":
+    PARSER = argparse.ArgumentParser()
+    PARSER.add_argument("files", nargs="+", help="The files to process")
+    ARGS = PARSER.parse_args()
+
+    process_metric(ARGS)
+
+    if METRIC_FAIL:
+        sys.exit(-1)
```

Comment:
Why not have a metric function array and then a cycle through it.

```suggestion
   metrics = [ ]
   for process_metric in metrics:
       process_metric(ARGS)
       if METRIC_FAIL:
           sys.exit(-1)
```

Then, in the future, for each metric's processing function, just add i to the array.

Reply:
That is a more flexible way to implement the checker flow. I just followed your idea, thanks!